components/omega/doc/design/PGrad.md

```diff
@@ -0,0 +1,224 @@
+```c++
+enum class PressureGradType{
+   Centered,
+   HighOrder
+}
+```
+The functions to compute the centered and high order pressure gradient terms will be implemented as functors and the pressure gradient class will have private instances of these classes.
+### 4.2 Methods
+
+```c++
+class PressureGrad{
+    public:
+        static PressureGrad *init();
+        static PressureGrad *get();
+        void computePressureGrad();
+    private:
+
+}
+```
+#### 4.2.1 Creation
+
+The constructor will be responsible for storing any static mesh information as private variables and handling the selection of the user-specified pressure gradient option.
+```c++
+PressureGrad::PressureGrad(const HorzMesh *Mesh, int NVertLevels, Config *Options);
+```
+
+#### 4.2.2 Initialization
+
+The init method will create the pressure gradient and return an error code:
+```c++
+int PressureGrad::init();
+```
+It calls the constructor to create a new pressure gradient instance, producing a static managed (unique) pointer to the single instance.
+
+#### 4.2.3 Retrieval
+
+There will be a method for getting a pointer to the pressure gradient instance:
+```c++
+PressureGrad *PressureGrad::get();
+```
+
+#### 4.2.4 Computation
+
+The public `computePressureGrad` method will rely on private methods for each specific pressure gradient option (centered and high order).
+Note that the functors called by `computePressureGrad` are responsible for computing the sum of the pressure gradient and geopotential gradient accumulated in the `Tend` output array.
+```c++
+void PressureGrad::computePressureGrad(const Array2DReal &Tend,
+                                       const Array2DReal &Pressure,
+                                       const Array2DReal &Geopotential,
+                                       const Array2DReal &SpecVol) {
+OMEGA_SCOPE(LocCenteredPGrad, CenteredPGrad)
+OMEGA_SCOPE(LocHighOrderPGrad, HighOrderPGrad)
+    if (PressureGradChoice == PressureGradType::Centered){
+       parallelFor("pgrad-centered", {NCellsAll, NChunks},
+       KOKKOS_LAMBDA(int ICell, int KChunk) {
+          LocCenteredPGrad(Tend, Pressure, Geopotential, SpecVol);
+       });
+    }
+    else if (PressureGradChoice == PressureGradType::HighOrder){
+       parallelFor("pgrad-highorder", {NCellsAll, NChunks},
+       KOKKOS_LAMBDA(int ICell, int KChunk) {
+          LocHighOrderPGrad(Tend, Pressure, Geopotential, SpecVol);
+        });
+    }
+```
+
+#### 4.2.5 Destruction and removal
+
+No operations are needed in the destructor.
+The clear method will remove the pressure gradient instance, calling the destructor in the process.
+```c++
+void PressureGrad::clear();
+```
+
+## Verification and Testing
+
+### Test: Spatial convergence to exact solution
+For a given analytical $v$,  $h$, and $b$, the spatial convergence of the pressure gradient can be assessed by computing errors on progressively finer meshes.
```

Comment:
What is $b$ in this case?

Reply:
I think maybe this should be fleshed out a bit more.  What analytic test would we use?